pkg/ebpf/c/tracee.bpf.c

```diff
@@ -7311,17 +7313,19 @@ static __always_inline bool is_socket_supported(struct socket *sock)
 
 // cgroupctxmap
 
-#define SUB_NET_PACKET_IP      (1 << 0) // submit net_packet_ip_base
-#define SUB_NET_PACKET_TCP     (1 << 1) // submit net_packet_tcp_base
-#define SUB_NET_PACKET_UDP     (1 << 2) // submit net_packet_udp_base
-#define SUB_NET_PACKET_ICMP    (1 << 3) // submit net_packet_icmp_base
-#define SUB_NET_PACKET_ICMPV6  (1 << 4) // submit net_packet_icmpv6_base
-#define SUB_NET_PACKET_DNS     (1 << 5) // submit net_packet_dns_base
+#define SUB_NET_PACKET_IP               (1 << 0) // submit net_packet_ip_base
+#define SUB_NET_PACKET_TCP              (1 << 1) // submit net_packet_tcp_base
+#define SUB_NET_PACKET_UDP              (1 << 2) // submit net_packet_udp_base
+#define SUB_NET_PACKET_ICMP             (1 << 3) // submit net_packet_icmp_base
+#define SUB_NET_PACKET_ICMPV6           (1 << 4) // submit net_packet_icmpv6_base
+#define SUB_NET_PACKET_DNS              (1 << 5) // submit net_packet_dns_base
+#define SUB_NET_PACKET_HTTP_REQUEST     (1 << 6) // submit net_packet_http_request_base
```

Comment:
Don't you think we should have SUB_NET_PACKET_HTTP here only ? Just like the DNS, the distinction between REQUEST/RESPONSE would be done in the derivation layer. WDYT ?

Reply:
yes i wanted it to be like the DNS - the problem is that the `http` package i'm using in the derivation layer expects me to know whether it's a request or a reponse. 
i'm checking if its a request or response in the kernel. so i think we can solve it by adding an argument of the direction of the packet to the HTTP base event, and then i'll know how to derive. sounds good?

Reply:
Hum... I did something similar to differentiate IPv4 and IPv6.. if you look at `cgroup_bpf_run_filter_skb` kprobe, I'm using "eventctx->retval = family", so userland knows what to expect (v4 or v6) in the derivation phase without adding anything to the context.

We could expand that idea (have an enum with multiple values for retval ?)... 

The idea of having more arguments to the base network event is also ok, I considered it in the past but wanted to avoid it as much as we could (to make uniform to all network base events just to have a single "payload" argument).

Adding more arguments to the event will require:

```C
typedef struct net_event_context {
    event_context_t eventctx;
    struct { // event arguments (needs packing), use anonymous struct to ...
        u8 index0;
        u32 bytes;
        u8 index1; // == 1, set by default
        u32 bytes;
        // ... (payload sent by bpf_perf_event_output)
    } __attribute__((__packed__)); // ... avoid address-of-packed-member warns
    // members bellow this point are metadata (not part of event to be sent)
    net_event_contextmd_t md;
} net_event_context_t;
```

and padding recalculation (need to confirm sizeof_net_event_context_t() function is good). 

I think I'd rather try to use regular existing fields (like the retval) with different enums... we could have something like:

#define XXX_ipv4 (1 << 0)
#define XXX_ipv6 (1 << 1)
#define XXX_HTTP_req (1 << 2)
#define XXX_HTTP_resp (1 << 3)

and set the retval to it. WDYT ?

Reply:
that's a nice approach. for now i think retval is good enough for our needs (maybe in the future we can add another argument).

i've updated this PR - it now checks whether HTTP request/response both in BPF part and in userpsace part.

once the types PR (#2535) will be merged i'll update this one to use the retval instead of having the same check twice.
(this one is currently for reference).